jung-visualization/src/main/java/edu/uci/ics/jung/visualization/subLayout/GraphCollapser.java

```diff
@@ -71,76 +73,110 @@ public Network collapse(Network inGraph, Network clusterGraph) {
     return graph;
   }
 
-  public Network expand(Network inGraph, Network clusterGraph) {
-    MutableNetwork graph = graphBuilder.build();
-    Collection clusterNodes = clusterGraph.nodes();
-    logger.fine("cluster to expand is " + clusterNodes);
-
-    // put all clusterGraph vertices and edges into the new Graph
-    for (Object node : clusterNodes) {
-      graph.addNode(node);
-      for (Object edge : clusterGraph.incidentEdges(node)) {
-        Object u = clusterGraph.incidentNodes(edge).nodeU();
-        Object v = clusterGraph.incidentNodes(edge).nodeV();
-        graph.addEdge(u, v, edge);
+  public Network expand(Network originalNetwork, Network inGraph, Network clusterGraphNode) {
```

Comment:
I suspect that 'expand' has always been 'broken' and worked only because of a fortunate ordering of the vertices in the old graph implementations